packages/functions/src/dal/AppointmentDataAccessLayer.ts

```diff
@@ -177,3 +177,11 @@ export function removeDonorFromDbAppointment(
     status: AppointmentStatus.AVAILABLE,
   };
 }
+
+export function confirmArrivedFromDbAppointment(
+  appointment: DbAppointment
+): DbAppointment {
+  appointment.donationDoneTimeMillis = admin.firestore.Timestamp.now();
+  appointment.status = AppointmentStatus.CONFIRMED;
```

Comment:
Wait, are we talking here about CONFIRMED (i.e user said they will come) or COMPLETED (i.e. user said they came and donated)?

Maybe we should also rename the API to make this clearer